### PR TITLE
Add warning for future change of argument name for `timeseries` from `asel` -> `atomgroup`

### LINKED ISSUE #3911: Change name of `asel` in timeseries to something more comprehensible.
I know we're just doing the same as what memoryreader does, but it just struck me that `asel` is such a weird parameter name for an atomgroup. Not much we can do here, but if I'm not the only one maybe we can consider changing this for 3.0 (in a different PR, let's get this merged!).

_Originally posted by @IAlibay in https://github.com/MDAnalysis/mdanalysis/pull/3890#discussion_r1014879026_
     

I agree with @IAlibay that the name `asel` is a little confusing and could do with a change in 3.0. 
leave your opinions below. 

Will require a deprecation warning first if we go ahead.

### PR DESCRIPTION
**Pull request by hmacdope**
Fixes #3911 

Changes made in this Pull Request:
 - Adds warning for future change of name in `timeseries` from `asel` to `atomgroup`. 
 

PR Checklist
------------
 - [X] Tests?
 - [X] CHANGELOG updated?
 - [X] Issue raised/referenced?

## Developers certificate of origin
- [X] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
:books: Documentation preview :books:: https://mdanalysis--4343.org.readthedocs.build/en/4343/

### CODE REVIEW

File: testsuite/MDAnalysisTests/coordinates/test_dcd.py [hunk 1]
```diff
@@ -236,7 +236,12 @@ def test_timeseries_empty_selection(universe_dcd):
         asel = universe_dcd.select_atoms('name FOO')
         universe_dcd.trajectory.timeseries(asel=asel)
 
-
+def test_timeseries_asel_warns_deprecation(universe_dcd):
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/coordinates/test_dcd.py]:
```suggestion

def test_timeseries_asel_warns_deprecation(universe_dcd):
```
pep8?

File: testsuite/MDAnalysisTests/coordinates/test_dcd.py [hunk 2]
```diff
@@ -236,7 +236,12 @@ def test_timeseries_empty_selection(universe_dcd):
...
+    allframes = universe_dcd.trajectory.timeseries(order='afc')
+    asel = universe_dcd.atoms[[1, 2, 3, 4]]
+    with pytest.warns(DeprecationWarning, match="asel argument to"):
+        xyz = universe_dcd.trajectory.timeseries(asel=asel, order='afc')
+
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/coordinates/test_dcd.py]:
```suggestion


```

File: package/MDAnalysis/coordinates/DCD.py [hunk 3]
```diff
@@ -315,6 +315,11 @@ def timeseries(self,
             ValueError now raised instead of NoDataError for empty input
             AtomGroup
         """
+        if asel:
```
Comment (IAlibay) [inline package/MDAnalysis/coordinates/DCD.py]: Can you add a deprecated entry and then also allow for the new keyword please?

File: package/MDAnalysis/coordinates/base.py [hunk 4]
```diff
@@ -1023,6 +1023,12 @@ def timeseries(self, asel: Optional['AtomGroup']=None,
 
         .. versionadded:: 2.4.0
         """
+        if asel:
```
Comment (IAlibay) [inline package/MDAnalysis/coordinates/base.py]: Same as above, we should support both keyword arguments and docs should reflect this.

File: package/MDAnalysis/coordinates/memory.py [hunk 5]
```diff
@@ -525,6 +525,12 @@ def timeseries(self, asel=None, start=0, stop=-1, step=1, order='afc'):
             ValueError now raised instead of NoDataError for empty input
             AtomGroup
         """
+        if asel:
```
Comment (IAlibay) [inline package/MDAnalysis/coordinates/memory.py]: As above

**Review by IAlibay — CHANGES_REQUESTED** [review]
Cheers, couple of things though.

**Comment by hmacdope** [comment]
@IAlibay still WIP ill ping for review when ready

**Comment by RMeli** [comment]
@hmacdope given your comment above I converted it to a draft for now.

File: package/MDAnalysis/coordinates/DCD.py (hunk 3 above)
Comment (hmacdope) [reply package/MDAnalysis/coordinates/DCD.py]: Done

File: package/MDAnalysis/coordinates/memory.py (hunk 5 above)
Comment (hmacdope) [reply package/MDAnalysis/coordinates/memory.py]: Done :)

**Comment by hmacdope** [comment]
@IAlibay @RMeli assuming tests pass I think this is ready for another review.

**Review by RMeli — APPROVED** [review]

**Review by IAlibay — APPROVED** [review]
Cheers

**Pull request merged**